src/dbnode/encoding/iterators_types.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package encoding
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	errIterateEqualTimestampStrategyNotSpecified = errors.New("iterate equal timestamp strategy not specified")
+)
+
+// IterateEqualTimestampStrategy describes the strategy of which value to
+// select when values with equal timestamps appear in the list of iterators.
+type IterateEqualTimestampStrategy uint8
+
+const (
+	// IterateLastPushed is useful for within a single replica, using the last
+	// immutable buffer that was created to decide which value to choose.
```

Comment:
The iterators have to be given in the correct order though right? Maybe clarify?

Reply:
Sure thing.